openpmd_viewer/openpmd_timeseries/data_reader/data_reader.py

```diff
@@ -0,0 +1,240 @@
+"""
+This file is part of the openPMD-viewer
+
+It routes the calls to the data reader to either
+the h5py data reader, or to openpmd-api.
+
+Copyright 2020, openPMD-viewer contributors
+Authors: Remi Lehe
+License: 3-Clause-BSD-LBNL
+"""
+from . import h5py_reader
+#from . import openpmd_api_reader
+
+
+class DataReader( object ):
+    """
+    TODO
+    """
+
+    def __init__(self, backend='h5py'):
+        """
+        TODO
+        """
+        self.backend = backend
+
+        # Point to the correct reader module
+        if self.backend == 'h5py':
+            self.iteration_to_file = {}
+
+#        elif self.backend == 'openpmd-api':
+#            self.reader_module = openpmd_api_reader
+        else:
+            raise RuntimeError('Unknown backend: %s' % backend)
+
+    def list_iterations(self, path_to_dir):
+        """
+        Return a list of the iterations that correspond to the files
+        in this directory. (The correspondance between iterations and
+        files is stored internally.)
+
+        Parameter
+        ---------
+        path_to_dir : string
+            The path to the directory where the hdf5 files are.
+
+        Returns
+        -------
+        an array of integers which correspond to the iteration of each file
+        (in sorted order)
+        """
+        if self.backend == 'h5py':
+            iterations, iteration_to_file = \
+                h5py_reader.list_files( path_to_dir )
+            # Store dictionary of correspondence between iteration and file
+            self.iteration_to_file = iteration_to_file
```

Comment:
could we store this in the `h5py_reader` itself?

Reply:
Right now `h5py_reader` is a module, so we could not store `iteration_to_file` in it. (e.g. if we create multiple `OpenPMDTimeSeries` in the same Python session, than these would need separate `iteration_to_file`, which would not work if that attribute is store in the module.)

Alternatively, we could create a dedicate `H5PyReader` object to store `iteration_to_file`, but that would be maybe a bit overkill.

Reply:
ok, good enough for now.